src/lib/storage/table.cpp

```diff
@@ -1,84 +1,106 @@
 #include "table.hpp"
 
+#include "resolve_type.hpp"
 #include "utils/assert.hpp"
+#include "value_segment.hpp"
 
 namespace opossum {
 
-Table::Table(const ChunkOffset target_chunk_size) {
-  // Implementation goes here
+Table::Table(const ChunkOffset target_chunk_size) : _target_chunk_size(target_chunk_size) {
+  create_new_chunk();
 }
 
 void Table::add_column_definition(const std::string& name, const std::string& type, const bool nullable) {
-  // Implementation goes here
+  _column_names.emplace_back(name);
+  _column_types.emplace_back(type);
+  _column_nullables.emplace_back(nullable);
 }
 
 void Table::add_column(const std::string& name, const std::string& type, const bool nullable) {
-  // Implementation goes here
-  Fail("Implementation is missing.");
+  DebugAssert(_chunks.size() <= 1, "It is only possible to add new columns to an empty table.");
```

Comment:
I think you should test whether the first chunk contains an element. Right now, the first chunk might be half full and you could still add a column. 
```suggestion
  DebugAssert(_chunks[0].size() == 0, "It is only possible to add new columns to an empty table.");
```